packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterPanel.tsx

```diff
@@ -87,12 +87,7 @@ const GridFilterPanel = React.forwardRef<HTMLDivElement, GridFilterPanelProps>(
       ...other
     } = props;
 
-    const applyFilter = React.useCallback(
-      (item: GridFilterItem) => {
-        apiRef.current.upsertFilterItem(item);
-      },
-      [apiRef],
-    );
+    const applyFilter = apiRef.current.upsertFilterItem;
 
```

Comment:
Overall I feel our codebase is a bit more complex than it needs to be. There's a bit of refactoring/simplification we could do with a lot of these component methods. E.g. this one is passing `apiRef.current.upsertFilterItem` around into a prop downstream, and it gets renamed to `applyFilter`, `applyFilterChange` and `applyValue` by the various children; we could simplify by just using the API.

Reply:
Agree, feel free to simplify things if you see such opportunities 👍

Reply:
The annoying part is that the props are all public so this refactoring can only be fully completed at V7 :| We should probably allocate some time for refactoring before the release. Or release more often.

Reply:
Could you create an issue with a list of changes and link it in https://github.com/mui/mui-x/issues/7902?